src/Workspaces/Core/Portable/Classification/SyntaxClassification/AbstractSyntaxClassificationService.Worker.cs

```diff
@@ -56,17 +55,20 @@ internal partial class AbstractSyntaxClassificationService
 
             internal static void Classify(
                 SemanticModel semanticModel,
-                TextSpan textSpan,
+                ImmutableArray<TextSpan> textSpans,
                 SegmentedList<ClassifiedSpan> list,
                 Func<SyntaxNode, ImmutableArray<ISyntaxClassifier>> getNodeClassifiers,
                 Func<SyntaxToken, ImmutableArray<ISyntaxClassifier>> getTokenClassifiers,
                 ClassificationOptions options,
                 CancellationToken cancellationToken)
             {
-                using var worker = new Worker(semanticModel, textSpan, list, getNodeClassifiers, getTokenClassifiers, options, cancellationToken);
+                foreach (var textSpan in textSpans)
+                {
+                    using var worker = new Worker(semanticModel, textSpan, list, getNodeClassifiers, getTokenClassifiers, options, cancellationToken);
 
-                worker._pendingNodes.Push(worker._syntaxTree.GetRoot(cancellationToken));
-                worker.ProcessNodes();
+                    worker._pendingNodes.Push(worker._syntaxTree.GetRoot(cancellationToken));
```

Comment:
>worker._syntaxTree.GetRoot(cancellationToken) [](http://example.com/codeflow?start=46&length=45)

nit: could push this outside the loop:

var root = semanticModel.SyntaxTree.GetRoot(cancellationToken);